xr-active-switch.ts

```diff
@@ -0,0 +1,115 @@
+import {Component, Object3D, property} from '@wonderlandengine/api';
+
+export class XrActiveSwitch extends Component {
+    static TypeName = 'xr-active-switch';
+
+    /** Action to perform when XR session starts */
+    @property.enum(['Activate', 'Deactivate', 'Toggle', 'Keep'])
+    ifInXR: string | number = 0;
+
+    /** Action to perform when XR session ends */
+    @property.enum(['Activate', 'Deactivate', 'Toggle', 'Keep'])
+    ifNotInXR: string | number = 1;
+
+    /** Scope of elements to affect */
+    @property.enum(['This Object', 'Children Only', 'This Object & Children'])
+    scope: string | number = 2;
+
+    private components: Component[] = [];
+    private onSessionStart!: () => void;
+    private onSessionEnd!: () => void;
+
+    start() {
+        this.components = [];
+        this.collectComponents();
+
+        // Initial state setup
+        if (this.engine.xr?.session) {
+            this.applyAction(this.ifInXR);
+        } else {
+            this.applyAction(this.ifNotInXR);
+        }
+
+        // Bind event handlers
+        this.onSessionStart = this.handleSessionStart.bind(this);
+        this.onSessionEnd = this.handleSessionEnd.bind(this);
```

Comment:
Use arrow notation functions for handleSessionEnd instead, see comment there. Then there is no need to .bind(this)